core/src/main/java/it/pagopa/selfcare/mscore/core/OnboardingDao.java

```diff
@@ -204,6 +249,24 @@ private void updateOrCreateUser(List<String> toUpdate, UserToOnboard userToOnboa
         }
     }
 
+    private Optional<OnboardedProduct> updateOrCreateUserWithState(List<String> toUpdate, UserToOnboard userToOnboard, Institution institution, OnboardingRequest request, String tokenId, RelationshipState state) {
+        try {
+            OnboardedUser onboardedUser = isNewUser(toUpdate, userToOnboard.getId());
```

Comment:
_naming_: for the "casual reader" `toUpdate` does not carry enough information about what it represents. probably a more meaningful name would help here (ie. something as expressive as `userToOnboard` which is a good name instead).

Reply:
we chose to postpone some refactors such as naming to avoid breaking some features